src/react/components/pages/editorPage/canvasHelpers.ts

```diff
@@ -27,6 +27,89 @@ export default class CanvasHelpers {
         }
     }
 
+    /**
+     * Adds region to regions if missing,
+     * Removes region from regions if contained
+     */
+    public static toggleRegion(regions: IRegion[], region: IRegion): void {
+        const index = regions.findIndex((r) => r.id === region.id);
+        if (index === -1) {
+            regions.push(region);
+        } else {
+            regions.splice(index, 1);
+        }
+    }
+
+    /**
+     * Finds a string in array of strings
+     */
+    public static find(items: string[], item: string): string {
+        return items.find((t) => t === item);
+    }
+
+    /**
+     * Finds region with id if contained in array of regions
+     */
+    public static findRegion(regions: IRegion[], id: string): IRegion {
+        return regions.find((r) => (r.id === id));
+    }
+
+    /**
+     * Find index of string in array of strings
+     */
+    public static findIndex(tags: string[], tag: string): number {
+        return tags.findIndex((t) => t === tag);
+    }
+
+    /**
+     * Adds tag to tags if not contained
+     */
+    public static addIfMissing(tags: string[], tag: string): void {
+        if (!CanvasHelpers.find(tags, tag)) {
+            tags.push(tag);
+        }
+    }
+
+    /**
+     * Adds all target tags if missing from tags
+     */
+    public static addAllIfMissing(tags: string[], targets: string[]): void {
```

Comment:
What is `targets` here?  Region ids? tags?

Reply:
Tags. I'll change the name and add docs